server/src/test/java/org/apache/druid/server/coordinator/rules/LoadRuleTest.java

```diff
@@ -685,6 +685,210 @@ public void testMaxLoadingQueueSize()
+            "tier2",
+            Stream.of(
+                holder3,
+                holder4
+            ).collect(Collectors.toCollection(() -> new TreeSet<>()))
+        )
+    );
+
+    CoordinatorStats stats = rule.run(
+        null,
+        DruidCoordinatorRuntimeParams.newBuilder()
+                                     .withDruidCluster(druidCluster)
+                                     .withSegmentReplicantLookup(SegmentReplicantLookup.make(druidCluster))
+                                     .withReplicationManager(throttler)
+                                     .withBalancerStrategy(mockBalancerStrategy)
+                                     .withBalancerReferenceTimestamp(DateTimes.of("2013-01-01"))
+                                     .withAvailableSegments(Collections.singletonList(segment)).build(),
+        segment
+    );
+
+    Assert.assertEquals(1L, stats.getTieredStat(LoadRule.ASSIGNED_COUNT, "tier1"));
+    Assert.assertEquals(2L, stats.getTieredStat(LoadRule.ASSIGNED_COUNT, "tier2"));
+
+    EasyMock.verify(throttler, mockPeon1, mockPeon2, mockPeon3, mockPeon4, mockBalancerStrategy);
+  }
+
+
+  /**
+   * 2 servers with a segment, one server in maintenance mode.
+   * Should drop a segment from both.
+   */
+  @Test
+  public void testDropDuringMaintenance()
+  {
+    final LoadQueuePeon mockPeon = createEmptyPeon();
+    mockPeon.dropSegment(EasyMock.anyObject(), EasyMock.anyObject());
+    EasyMock.expectLastCall().times(2);
+    EasyMock.expect(mockBalancerStrategy.pickServersToDrop(EasyMock.anyObject(), EasyMock.anyObject()))
+            .andDelegateTo(balancerStrategy)
+            .times(2);
+    EasyMock.replay(throttler, mockPeon, mockBalancerStrategy);
+
+    LoadRule rule = createLoadRule(ImmutableMap.of(
+        "tier1", 0
+    ));
+
+    final DataSegment segment1 = createDataSegment("foo1");
+    final DataSegment segment2 = createDataSegment("foo2");
+
+    DruidServer server1 = createServer("tier1");
+    server1.addDataSegment(segment1);
+    DruidServer server2 = createServer("tier1");
+    server2.addDataSegment(segment2);
+
+    DruidCluster druidCluster = new DruidCluster(
+        null,
+        ImmutableMap.of(
+            "tier1",
+            Stream.of(
+                new ServerHolder(
+                    server1.toImmutableDruidServer(),
+                    mockPeon,
+                    true
+                ),
+                new ServerHolder(
+                    server2.toImmutableDruidServer(),
+                    mockPeon,
+                    false
+                )
+            ).collect(Collectors.toCollection(() -> new TreeSet<>(Collections.reverseOrder())))
+        )
+    );
+
+    DruidCoordinatorRuntimeParams params = DruidCoordinatorRuntimeParams.newBuilder()
+                                                                       .withDruidCluster(druidCluster)
+                                                                       .withSegmentReplicantLookup(
```

Comment:
Could break `.newBuilder()` on the next line to avoid this far right alignment

Reply:
:white_check_mark: